src/main/java/org/folio/inventory/resources/MoveApi.java

```diff
@@ -0,0 +1,143 @@
+
+import org.apache.commons.collections15.ListUtils;
+import org.folio.inventory.common.WebContext;
+import org.folio.inventory.domain.items.Item;
+import org.folio.inventory.domain.items.ItemCollection;
+import org.folio.inventory.storage.Storage;
+import org.folio.inventory.storage.external.CollectionResourceClient;
+import org.folio.inventory.storage.external.CqlQuery;
+import org.folio.inventory.storage.external.MultipleRecordsFetchClient;
+import org.folio.inventory.support.ItemUtil;
+import org.folio.inventory.support.http.client.OkapiHttpClient;
+import org.folio.inventory.support.http.server.JsonResponse;
+import org.folio.inventory.support.http.server.ServerErrorResponse;
+import org.folio.inventory.support.http.server.ValidationError;
+
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+
+public class MoveApi extends AbstractInventoryResource {
+
+  public static final String TO_HOLDINGS_RECORD_ID = "toHoldingsRecordId";
+  public static final String ITEM_IDS = "itemIds";
+
+  public MoveApi(final Storage storage, final HttpClient client) {
+    super(storage, client);
+  }
+
+  @Override
+  public void register(Router router) {
+    router.post("/inventory/items/move")
+      .handler(this::moveItems);
+  }
+
+  private void moveItems(RoutingContext routingContext) {
+
+    WebContext context = new WebContext(routingContext);
+    JsonObject itemsMoveJsonRequest = routingContext.getBodyAsJson();
+
+    Optional<ValidationError> validationError = itemsMoveHasRequiredFields(itemsMoveJsonRequest);
+    if (validationError.isPresent()) {
+      unprocessableEntity(routingContext.response(), validationError.get());
+    } else {
+      String toHoldingsRecordId = itemsMoveJsonRequest.getString(TO_HOLDINGS_RECORD_ID);
+      List<String> itemIdsToUpdate = toListOfStrings(itemsMoveJsonRequest.getJsonArray(ITEM_IDS));
+      storage.getHoldingsRecordCollection(context)
+        .findById(toHoldingsRecordId)
+        .thenAccept(holding -> {
+          if (Objects.nonNull(holding)) {
+            try {
+              OkapiHttpClient okapiClient = createHttpClient(routingContext, context);
+              CollectionResourceClient itemsStorageClient = createItemsStorageClient(okapiClient, context);
+              MultipleRecordsFetchClient multipleRecordsFetchClient = createItemsFetchClient(itemsStorageClient);
+
+              multipleRecordsFetchClient.find(itemIdsToUpdate, this::fetchItemsCql)
+                .thenAccept(jsons -> {
+                  List<Item> itemsToUpdate = jsons.stream()
+                    .map(ItemUtil::fromStoredItemRepresentation)
+                    .map(item -> item.withHoldingId(toHoldingsRecordId))
+                    .collect(toList());
+                  updateItems(routingContext, context, itemIdsToUpdate, itemsToUpdate);
+                });
```

Comment:
What happens when this process fails? Is the intention that the `catch` statement below it will be executed?

Reply:
this catch for checked  exception: java.net.MalformedURLException from createHttpClient and createItemsStorageClient methods. I think server error will be returned without items updating.

Reply:
What happens if an exception is thrown within this block of code?

Reply:
ServerErrorResponse.internalError(routingContext.response(), e) will be executed and process will be stopped.

Reply:
How will that code be executed? By the `catch` statement below?

I'm asking because I don't think that exceptions thrown within `thenAccept` methods will throw the exception in the invoking context. Instead, they will provide a new `CompletionStage` that has failed exceptionally.

Reply:
Ah! OK. I have added exceptionally block for this possible exception. Thanks!